File: app/src/main/java/com/paycraft/ticketbooker/db/DBClient.java
package com.paycraft.ticketbooker.db;


import android.content.Context;

import androidx.room.Room;

public class DBClient {
    private static DBClient mInstance;

    //our app database object
    private AppDB appDatabase;

    private DBClient(Context mCtx) {

        appDatabase = Room.databaseBuilder(mCtx, AppDB.class, "TicketBooker").allowMainThreadQueries().build();
    }

    public static synchronized DBClient getInstance(Context mCtx) {
        if (mInstance == null) {
            mInstance = new DBClient(mCtx);
        }
        return mInstance;
    }

    public AppDB getAppDatabase() {
        return appDatabase;
    }
}
 

File: app/src/main/java/com/paycraft/ticketbooker/db/DBClient.kt
package com.paycraft.ticketbooker.db


import android.content.Context

import androidx.room.Room

//class DBClient private constructor(mCtx: Context) {
//
//    //our app database object
//    val appDatabase: AppDB
//
//    init {
//
//        appDatabase = Room.databaseBuilder(mCtx, AppDB::class.java, "TicketBooker").allowMainThreadQueries().build()
//    }
//
//    companion object :DBClientHolder<DBClient,Context>(::DBClient) {
//        private  var mInstance: DBClient? = null;
//
//        @Synchronized
//        fun getInstance(mCtx: Context): DBClient {
//            if (mInstance == null) {
//                mInstance = DBClient(mCtx)
//            }
//            return mInstance!!
//        }
//    }
//}
//
//open class DBClientHolder<T, U>(value: KFunction1<U, T>) {
//
//}
